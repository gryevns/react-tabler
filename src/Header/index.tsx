import React from "react";
import cn from "classnames";
import { Search } from "react-feather";

export interface HeaderTitleProps {
    children: React.ReactNode;
    href: string;
}

const HeaderTitle = ({ children }: HeaderTitleProps, href) => (
    <a
        href={href}
        className="navbar-brand navbar-brand-autodark d-none-navbar-vertical"
    >
        {children}
    </a>
);

const HeaderSearch = ({ placeholder = "Search..." }) => (
    <form className="form-inline w-50 mr-4 d-none d-md-flex ">
        <div className="input-icon w-100">
            <span className="input-icon-addon">
                <Search size={18} />
            </span>
            <input
                type="text"
                className="form-control form-control-flush w-100"
                placeholder={placeholder}
            />
        </div>
    </form>
);

export interface HeaderProps {
    children: React.ReactNode;
    className?: string;
}

const Header = ({ children, className }: HeaderProps) => {
    const classes = cn(
        "navbar navbar-light navbar-secondary navbar-expand",
        className,
    );
    return <nav className={classes}>{children}</nav>;
};

Header.Title = HeaderTitle;
Header.Search = HeaderSearch;

export default Header;
