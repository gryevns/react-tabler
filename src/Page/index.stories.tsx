import React from "react";
import { DollarSign, ShoppingCart, Users, Plus, MessageCircle } from "react-feather";

import Badge from "../Badge";
import Card from "../Card";
import Page from "../Page";
import { Heading } from "../Header/index.stories";
import { Navigation } from "../Nav/index.stories";
import Grid from "../Grid";
import Table from "../Table";
import ThemeProvider, { ThemeContext } from "../ThemeProvider";
import { SocialMediaTraffice, Tasks } from "../Table/index.stories";
import List from "../List";
import {
    ChangeStat,
    ChildrenStat,
    IconStat,
    PostCard,
    ProgressStat,
} from "../Card/index.stories";
import { Responsive as UserTable } from "../Table/index.stories";

import { Browsers } from "../Table/index.stories";
import Avatar from "../Avatar";
import Progress from "../Progress";
import Buttons from "../Buttons";

export default {
    title: "Page",
};

const Projects = () => (
    <Card>
        <Card.Header>
            <Card.Title>Projects</Card.Title>
        </Card.Header>
        <Table cards>
            <Table.Body>
                <Table.Row>
                    <Table.Col>Admin Template</Table.Col>
                    <Table.Col className="text-right">
                        <Badge color="gray">65%</Badge>
                    </Table.Col>
                </Table.Row>
                <Table.Row>
                    <Table.Col>Landing Page</Table.Col>
                    <Table.Col className="text-right">
                        <Badge color="green">Finished</Badge>
                    </Table.Col>
                </Table.Row>
                <Table.Row>
                    <Table.Col>Backend UI</Table.Col>
                    <Table.Col className="text-right">
                        <Badge color="red">Rejected</Badge>
                    </Table.Col>
                </Table.Row>
                <Table.Row>
                    <Table.Col>Personal Blog</Table.Col>
                    <Table.Col className="text-right">
                        <Badge color="gray">40%</Badge>
                    </Table.Col>
                </Table.Row>
                <Table.Row>
                    <Table.Col>Email Templates</Table.Col>
                    <Table.Col className="text-right">
                        <Badge color="gray">13%</Badge>
                    </Table.Col>
                </Table.Row>
                <Table.Row>
                    <Table.Col>Corporate Website</Table.Col>
                    <Table.Col className="text-right">
                        <Badge color="yellow">Pending</Badge>
                    </Table.Col>
                </Table.Row>
            </Table.Body>
        </Table>
    </Card>
);

const AvatarCard = () => (
    <Card>
        <Card.Body className="p-4 py-5 text-center">
            <Avatar size="xl" className="mb-4">
                W
            </Avatar>
            <h3 className="mb-0">New website</h3>
            <p className="text-muted">Due to: 28 Aug 2019</p>
            <p className="mb-3">
                <Badge color="red" soft>
                    Waiting
                </Badge>
            </p>
            <Avatar.List>
                <Avatar imageURL="" />
                <Avatar>JL</Avatar>
                <Avatar imageURL="" />
                <Avatar imageURL="" />
                <Avatar imageURL="" />
            </Avatar.List>
        </Card.Body>
        <Progress card>
            <Progress.Bar width={38} />
        </Progress>
    </Card>
);

export const Default = () => {
    return (
        <ThemeProvider>
            <Page>
                <Navigation />
                <Heading />
                <Page.Content>
                    <Page.Header>
                        <div className="row align-items-center">
                            <div className="col-auto">
                                <Page.Pretitle>Overview</Page.Pretitle>
                                <Page.Title>Dashboard</Page.Title>
                            </div>
                            <div className="col-auto ml-auto d-print-none">
                                <span className="d-none d-sm-inline">
                                    <ThemeContext.Consumer>
                                        {({ theme, toggleTheme }) => (
                                            <>
                                                <Buttons.Button
                                                    variant="secondary"
                                                    onClick={toggleTheme}
                                                >
                                                    {theme === "light"
                                                        ? "Dark Theme"
                                                        : "Light Theme"}
                                                </Buttons.Button>
                                            </>
                                        )}
                                    </ThemeContext.Consumer>
                                </span>
                                <Buttons.Button
                                    icon
                                    variant="primary"
                                    className="ml-3 d-sm-inline-block"
                                >
                                    <Plus size={18} />
                                    &nbsp; Create new report
                                </Buttons.Button>
                            </div>
                        </div>
                    </Page.Header>
                    <Grid.Row cards deck>
                        <Grid.Col md={3} sm={6}>
                            <ChangeStat
                                movement={6}
                                total="43"
                                label="New Tickets"
                            />
                        </Grid.Col>
                        <Grid.Col md={3} sm={6}>
                            <ChangeStat
                                movement={-3}
                                total="17"
                                label="Closed Today"
                            />
                        </Grid.Col>
                        <Grid.Col md={3} sm={6}>
                            <ProgressStat
                                label="New Feedback"
                                color="red"
                                width={28}
                                value="62"
                            />
                        </Grid.Col>
                        <Grid.Col md={3} sm={6}>
                            <ProgressStat
                                label="Today Profit"
                                color="green"
                                width={84}
                                value="$652"
                            />
                        </Grid.Col>
                    </Grid.Row>
                    <Grid.Row cards deck>
                        <Grid.Col sm={6} lg={3}>
                            <IconStat
                                total="132"
                                label="Sales"
                                value="12 waiting payments"
                            >
                                <DollarSign color="white" />
                            </IconStat>
                        </Grid.Col>
                        <Grid.Col sm={6} lg={3}>
                            <IconStat
                                total="78"
                                label="Orders"
                                value="32 shipped"
                                color="green"
                            >
                                <ShoppingCart color="white" />
                            </IconStat>
                        </Grid.Col>
                        <Grid.Col sm={6} lg={3}>
                            <IconStat
                                total="1,352"
                                label="Members"
                                value="163 registered today"
                                color="red"
                            >
                                <Users color="white" />
                            </IconStat>
                        </Grid.Col>
                        <Grid.Col sm={6} lg={3}>
                            <IconStat
                                total="132"
                                label="Comments"
                                value="16 waiting"
                                color="yellow"
                            >
                                <MessageCircle color="white" />
                            </IconStat>
                        </Grid.Col>
                    </Grid.Row>
                    <Grid.Row cards deck>
                        <Grid.Col lg={6} sm={12}>
                            <PostCard />
                        </Grid.Col>
                        <Grid.Col lg={6} sm={12}>
                            <PostCard />
                        </Grid.Col>
                    </Grid.Row>
                    <Grid.Row>
                        <Grid.Col>
                            <UserTable />
                        </Grid.Col>
                    </Grid.Row>
                    <Grid.Row deck cards>
                        <Grid.Col sm={6} lg={4}>
                            <Browsers />
                        </Grid.Col>
                        <Grid.Col sm={6} lg={4}>
                            <Projects />
                        </Grid.Col>
                        <Grid.Col sm={12} lg={4}>
                            {/* <Members /> */}
                            <AvatarCard />
                        </Grid.Col>
                    </Grid.Row>
                    <Grid.Row deck cards>
                        <Grid.Col lg={3} sm={6}>
                            <ChildrenStat
                                movement={5}
                                label="Users Online"
                                total="423"
                                color="blue"
                            />
                        </Grid.Col>
                        <Grid.Col lg={3} sm={6}>
                            <ChildrenStat
                                movement={5}
                                label="Users Online"
                                total="423"
                                color="red"
                            />
                        </Grid.Col>
                        <Grid.Col lg={3} sm={6}>
                            <ChildrenStat
                                movement={5}
                                label="Users Online"
                                total="423"
                                color="green"
                            />
                        </Grid.Col>
                        <Grid.Col lg={3} sm={6}>
                            <ChildrenStat
                                movement={5}
                                label="Users Online"
                                total="423"
                                color="yellow"
                            />
                        </Grid.Col>
                    </Grid.Row>
                    <Grid.Row deck cards>
                        <Grid.Col width={12} lg={4}>
                            <SocialMediaTraffice />
                        </Grid.Col>
                        <Grid.Col width={12} lg={8}>
                            <Tasks />
                        </Grid.Col>
                    </Grid.Row>
                </Page.Content>
                <Page.SubFooter>
                    <Grid.Row>
                        <Grid.Col lg={8}>
                            <Grid.Row>
                                <Grid.Col width={6} md={3}>
                                    <List unstyled className="mb-0">
                                        <List.Item>
                                            <a href="#">First Link</a>
                                        </List.Item>
                                        <List.Item>
                                            <a href="#">Second Link</a>
                                        </List.Item>
                                    </List>
                                </Grid.Col>
                                <Grid.Col width={6} md={3}>
                                    <List unstyled className="mb-0">
                                        <List.Item>
                                            <a href="#">Third Link</a>
                                        </List.Item>
                                        <List.Item>
                                            <a href="#">Fourth Link</a>
                                        </List.Item>
                                    </List>
                                </Grid.Col>
                                <Grid.Col width={6} md={3}>
                                    <List unstyled className="mb-0">
                                        <List.Item>
                                            <a href="#">Fifth Link</a>
                                        </List.Item>
                                        <List.Item>
                                            <a href="#">Sixth Link</a>
                                        </List.Item>
                                    </List>
                                </Grid.Col>
                                <Grid.Col width={6} md={3}>
                                    <List unstyled className="mb-0">
                                        <List.Item>
                                            <a href="#">Seventh Link</a>
                                        </List.Item>
                                        <List.Item>
                                            <a href="#">Eigth Link</a>
                                        </List.Item>
                                    </List>
                                </Grid.Col>
                            </Grid.Row>
                        </Grid.Col>
                        <Grid.Col lg={4}>
                            Premium and Open Source dashboard template with
                            responsive and high quality UI. For Free!
                        </Grid.Col>
                    </Grid.Row>
                </Page.SubFooter>
                <Page.Footer>React Tabler</Page.Footer>
            </Page>
        </ThemeProvider>
    );
};
