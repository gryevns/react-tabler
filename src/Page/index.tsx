import React from "react";
import cn from "classnames";

import { NavContext } from "../Nav";

const Page = ({ children }) => {
    const [menuVisible, setMenuVisible] = React.useState(false);
    return (
        <NavContext.Provider value={{ menuVisible, setMenuVisible }}>
            <div className="page">{children}</div>
        </NavContext.Provider>
    );
};

const PageHeader = ({ children }) => <div className="page-header">{children}</div>;

const PagePreTitle = ({ children }) => (
    <div className="page-pretitle">{children}</div>
);

const PageTitle = ({ children }) => <h2 className="page-title">{children}</h2>;

const PageSubTitle = ({ children }) => (
    <div className="page-subtitle">{children}</div>
);

const PageOptions = ({ children }) => (
    <div className="page-options d-flex">{children}</div>
);

const Content = ({ children }) => <div className="content">{children}</div>;

const Footer = ({ children }) => <footer className="footer">{children}</footer>;

const SubFooter = ({ children }) => <div className="footer">{children}</div>;

interface ContainerProps {
    children: React.ReactNode;
    size?: "md" | "lg" | "xl" | "fluid";
}

const Container = ({ children, size }: ContainerProps) => {
    const classes = cn({ container: !size, [`container-${size}`]: size });
    return <div className={classes}>{children}</div>;
};

const Empty = ({ children }) => <div className="empty">{children}</div>;

Page.Header = PageHeader;
Page.Pretitle = PagePreTitle;
Page.Title = PageTitle;
Page.SubTitle = PageSubTitle;
Page.Options = PageOptions;
Page.Content = Content;
Page.Footer = Footer;
Page.SubFooter = SubFooter;
Page.Container = Container;
Page.Empty = Empty;

export default Page;
