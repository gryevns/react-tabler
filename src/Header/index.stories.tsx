import React from "react";

import Dropdown from "../Dropdown";
import Avatar from "../Avatar";
import Nav from "../Nav";
import Card from "../Card";
import Header from ".";
import Page from "../Page";
import Flag from "../Flags";
import { Bell, Settings } from "react-feather";

export default {
    title: "Header",
};

export const Heading = () => (
    <Header>
        <Page.Container>
            <Header.Title href=".">Tabler</Header.Title>
            <Header.Search />
            <Nav.Bar className="ml-auto">
                <Dropdown>
                    <Dropdown.Link>
                        <Settings size={15} />
                    </Dropdown.Link>
                    <Dropdown.Menu arrow position="right">
                        <Dropdown.Item>Action</Dropdown.Item>
                        <Dropdown.Item>Another action</Dropdown.Item>
                        <Dropdown.Divider />
                        <Dropdown.Item>Separated link</Dropdown.Item>
                    </Dropdown.Menu>
                </Dropdown>
                <Dropdown>
                    <Dropdown.Link>
                        <Flag country="us" className="mr-1" />
                        English
                    </Dropdown.Link>
                    <Dropdown.Menu arrow position="right">
                        <Dropdown.Item>
                            <Flag country="de" />
                            &nbsp; German
                        </Dropdown.Item>
                        <Dropdown.Item>
                            <Flag country="pl" />
                            &nbsp; Polish
                        </Dropdown.Item>
                        <Dropdown.Item>
                            <Flag country="jp" />
                            &nbsp; Japan
                        </Dropdown.Item>
                        <Dropdown.Item>
                            <Flag country="rs" />
                            &nbsp; Russian
                        </Dropdown.Item>
                    </Dropdown.Menu>
                </Dropdown>
                <Dropdown>
                    <Dropdown.Link>
                        <Bell size={15} />
                    </Dropdown.Link>
                    <Dropdown.Menu
                        arrow
                        card
                        position="right"
                        style={{ maxWidth: "20rem" }}
                    >
                        <Card>
                            <Card.Body>
                                Lorem ipsum dolor sit amet, consectetur adipisicing
                                elit. Accusamus ad amet consectetur exercitationem
                                fugiat in ipsa ipsum, natus odio quidem quod
                                repudiandae sapiente. Amet debitis et magni maxime
                                necessitatibus ullam.
                            </Card.Body>
                        </Card>
                    </Dropdown.Menu>
                </Dropdown>
                <Dropdown className="pl-3">
                    <Dropdown.Link>
                        <Avatar>GE</Avatar>
                        <div className="d-none d-lg-block pl-2">
                            <div>Pawe≈Ç Kuna</div>
                            <div className="mt-1 small text-muted">Founder</div>
                        </div>
                    </Dropdown.Link>
                    <Dropdown.Menu
                        arrow
                        card
                        position="right"
                        style={{ maxWidth: "20rem" }}
                    >
                        <Dropdown.Item>Action</Dropdown.Item>
                        <Dropdown.Item>Another action</Dropdown.Item>
                        <Dropdown.Divider />
                        <Dropdown.Item>Separated link</Dropdown.Item>
                    </Dropdown.Menu>
                </Dropdown>
            </Nav.Bar>
        </Page.Container>
    </Header>
);
