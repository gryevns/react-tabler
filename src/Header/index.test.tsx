import React from "react";
import { shallow } from "enzyme";

import Header from "./index";
import Page from "../Page";

describe("Header", () => {
    it("should match snapshot", () => {
        const wrapper = shallow(
            <Header>
                <Page.Container>
                    <Header.Title href="#">Title</Header.Title>
                </Page.Container>
            </Header>,
        );
        expect(wrapper).toMatchSnapshot();
    });
});
