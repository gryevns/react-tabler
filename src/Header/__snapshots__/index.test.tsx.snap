// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot 1`] = `
<nav
  className="navbar navbar-light navbar-secondary navbar-expand"
>
  <Container>
    <HeaderTitle
      href="#"
    >
      Title
    </HeaderTitle>
  </Container>
</nav>
`;
